### PR TITLE
Create task-annotation-generator tool script

### PR DESCRIPTION
**Pull request by WangWinson**
The tool script leverage `TaskOption.Validator` to `register` all task schemas,
then get all resolved schema to generate a `task-doc-data.json`, which will be using for task doc build script (later updated by @cgx027 )

@RackHD/corecommitters @cgx027 @iceiilin

### CODE REVIEW

**Comment by brianparry** [comment]
test this please

**Comment by iceiilin** [comment]
+1

File: task-annotation-generator.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                name: data.name + '_' + name,
+                title: data.title + '.' + name,
+                group: data.group,
+                groupTitle: data.groupTitle
+            };
+            subItems = subItems.concat(self.generateDocData(option, subTemp));
+        }
+
+        var fieldTemp = {
+            group: 'g1', // TODO: find out gourp usage
```
Comment (yyscamper) [inline task-annotation-generator.js]: typo: gourp -> group

File: task-annotation-generator.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+'use strict';
+
+var fs = require('fs');
+
+function TaskAnnotation(validator) {
+    this.validator = validator;
+}
+
+TaskAnnotation.prototype.mergeSchema = function (obj) {
```
Comment (yyscamper) [inline task-annotation-generator.js]: Could you please add some comment for each function to help other people to understand the design?

File: task-annotation-generator.js (hunk 1 above)
Comment (WangWinson) [inline task-annotation-generator.js]: oh, you catch me again.

File: task-annotation-generator.js (hunk 2 above)
Comment (WangWinson) [inline task-annotation-generator.js]: Sure. I can do it

**Comment by yyscamper** [comment]
How about adding some unit-test? though this file is not used by RackHD service code, but it contains some complex logic and some unit-test will prevent the task-annotation from being broken down.

**Comment by WangWinson** [comment]
@yyscamper 
Actually, this is a tool script, which should be a part of build script, instead of a part of deliver code.
I know I can add some unit test for it , since it's written in JS.

There is another story will need to enhance this script to make the doc task oriented instead of task schema oriented. Also, another PR for build script in on-http will have dependency on this PR. So, how about I make up some unit test in the later PR.

**Comment by yyscamper** [comment]
I agree this is just a kind of build script, previously we don't enforce unit-test for such kind of script, for example like some of scripts in on-http/data/templates, I also agree if your script is not written javascript, I will not give the suggestion about unit-testing ^^. So it's not a gating for me. It will be much better if you can still write some unit-test after your enhance story finished.
:+1:

**Pull request merged**